hil/ext/switches/juniper.py

```diff
@@ -0,0 +1,294 @@
+
+# Copyright 2013-2017 Massachusetts Open Cloud Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the
+# License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS
+# IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied.  See the License for the specific language
+# governing permissions and limitations under the License.
+"""A switch driver for the juniper QFX series.
+
+But in theory can work for any Juniper switch that supports NETCONF.
+Driver uses the python library PyEZ made available by Juniper.
+By default it sets NETCONF session over ssh, i.e. make RPC calls over ssh.
+For this to work, make sure that the Juniper switch is configured
+appropriately to recieve NETCONF connections. Its default port is 830.
+"""
+
+import re
+import logging
+import schema
+import json
+
+from jnpr.junos import Device
+from jnpr.junos.exception import ConnectError
+from jnpr.junos.utils.config import Config
+
+
+from hil.model import db, Switch
+from hil.migrations import paths
+from os.path import dirname, join
+from hil.migrations import paths
+from hil.ext.switches.junos.config_tables.ConfigTables import InterfaceConfigTable
+logger = logging.getLogger(__name__)
+paths[__name__] = join(dirname(__file__), 'migrations', 'juniper')
```

Comment:
it will be required if you wish to add Juniper switch to a running HIL installation.

Reply:
in that case you should generate a migration script too with this PR. 

also, as far as I know, running `hil-admin db create` creates any new tables so a migration script might not be needed. @zenhack can you confirm this?

Reply:
I tested it, if a new driver is introduced to a running HIL installation it does not require a migration script. To add the new table to the database, stop the HIL servers and simple run `hil-admin db create` again. It will create any new tables based on the changes to the `hil.cfg`